test/framework/src/main/java/org/elasticsearch/search/MockSearchService.java

```diff
@@ -59,13 +59,42 @@ public void onModule(SearchModule module) {
     private static final Map<SearchContext, Throwable> ACTIVE_SEARCH_CONTEXTS = new ConcurrentHashMap<>();
 
     /** Throw an {@link AssertionError} if there are still in-flight contexts. */
-    public static void assertNoInFLightContext() {
+    public static void assertNoInFlightContext() {
         final Map<SearchContext, Throwable> copy = new HashMap<>(ACTIVE_SEARCH_CONTEXTS);
         if (copy.isEmpty() == false) {
-            throw new AssertionError("There are still " + copy.size() + " in-flight contexts", copy.values().iterator().next());
+            Map.Entry<SearchContext, Throwable> firstOpen = copy.entrySet().iterator().next();
+            SearchContext context = firstOpen.getKey();
+            StringBuilder message = new StringBuilder().append(context.shardTarget());
+            if (context.searchType() != SearchType.DEFAULT) {
+                message.append("searchType=[").append(context.searchType()).append("]");
+            }
+            if (context.scrollContext() != null) {
+                message.append("scroll=[").append(context.scrollContext().scroll.keepAlive()).append("]");
+            }
+            message.append(" query=[").append(context.query()).append("]");
+            RuntimeException cause = new RuntimeException(message.toString());
+            cause.setStackTrace(firstOpen.getValue().getStackTrace());
```

Comment:
I just didn't want to build the message if we weren't sure we were going to use it. Is it worth it, given how often we make search contexts?